src/lua/uri.lua

```diff
@@ -226,7 +226,11 @@ local function format(uri, write_password)
     uribuf.scheme = uri.scheme
     uribuf.login = uri.login
     uribuf.password = uri.password
-    uribuf.host = uri.host
+    if uri.ipv6 ~= nil then
+        uribuf.host = string.format("[%s]", uri.host)
+    else
+        uribuf.host = uri.host
+    end
```

Comment:
I would implement in on the C layer, in the `uri_format` function. This way all the usages of `uri_format()` will gain correctly formatted URI. For example, this function is called for appliers to show them in `box.info` (and, maybe, in logs).

We can use the `host_hint == 2` condition for this purpose.

Reply:
corrected